integrations/jenkins/main.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any
+from loguru import logger
+from enum import StrEnum
+
+from client import JenkinsClient
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+class ObjectKind(StrEnum):
+    JOB = "job"
+    BUILD = "build"
+
+    @staticmethod
+    def get_object_kind_for_event(obj_type: str) -> str | None:
+        if obj_type.startswith("item"):
+            return ObjectKind.JOB
+        elif obj_type.startswith("run"):
+            return ObjectKind.BUILD
+        else:
+            return None
+
+
+def init_client() -> JenkinsClient:
+    return JenkinsClient(
+        ocean.integration_config["jenkins_host"],
+        ocean.integration_config["jenkins_user"],
+        ocean.integration_config["jenkins_token"],
+    )
+
+
+@ocean.on_resync()
+async def on_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    jenkins_client = init_client()
+
+    async for _jobs in jenkins_client.get_all_jobs_and_builds():
+        if kind == ObjectKind.JOB:
+            yield _jobs
```

Comment:
please remove the builds from these

Reply:
I don't understand this comment. 

Builds are located within jobs. When the resync is triggered for the `JOB` type, it retrieves the job information. When it processes the `BUILD` type, we retrieve the data from the cache.

Reply:
Yeah but rn you retrieve the builds also in the jobs data

Reply:
ooooh, i get it now.

Reply:
I have now separated them out.